code/asteroid/asteroid.cpp

```diff
@@ -262,6 +262,36 @@ static void inner_bound_pos_fixup(asteroid_field *asfieldp, vec3d *pos)
 	}
 }
 
+/**
+* Randomly select an asteroid subtype
+*/
+SCP_string pick_random_asteroid_type()
+{
+	// get a valid subtype
+	int counter = Random::next(Asteroid_field.num_used_field_debris_types);
+	SCP_string subtype = "";
+	for (size_t j = 0; j < Asteroid_field.field_asteroid_type.size(); j++) {
+		if (counter == 0) {
+			subtype = Asteroid_field.field_asteroid_type[j];
+			break;
+		} else
+			counter--;
+	}
+
+	return subtype;
+}
```

Comment:
Since `num_used_field_debris_types` is equal to `field_asteroid_type.size()`, this effectively just boils down to a much much simpler function with no loop:
```suggestion
const SCP_string& pick_random_asteroid_type()
{
	return Asteroid_field.field_asteroid_type[Random::next(Asteroid_field.field_asteroid_type.size())];
}
```